raco/algebra.py

```diff
@@ -1363,6 +1365,38 @@ def scheme(self):
         return self._scheme
 
 
+class SampleScan(ZeroaryOperator):
+
+    """Logical Sample Operator"""
+
+    def __init__(self, relation_key, _scheme, sample_size, with_replacement):
+        self.relation_key = relation_key
+        self._scheme = _scheme
+        self.sample_size = sample_size
+        self.with_replacement = with_replacement
+        self.sample_type = "WR" if self.with_replacement else "WoR"
```

Comment:
this field appears to be redundant with `self.with_replacement`. Can you just have a private method to return the appropriate one?